pegasus/sites.v3/code.org/public/pd-workshop-survey/counselor-admin/splat.haml

```diff
@@ -0,0 +1,186 @@
+---
+title: "Workshop Survey for Counselors and Administrators"
+---
+:ruby
+  dont_cache
+  enrollment_code = request.splat_path_info[1..-1]
+  enrollment = DASHBOARD_DB[:pd_enrollments].where(code: enrollment_code).first
+  pass unless enrollment
+
+  workshop = DASHBOARD_DB[:pd_workshops].where(id: enrollment[:pd_workshop_id]).first
+  pass unless workshop
+
+  teacher_id = enrollment[:user_id]
+  teacher = DASHBOARD_DB[:users].where(id: teacher_id).first
+  name = teacher.nil? ? enrollment[:name] : teacher[:name]
+  email = teacher.nil? ? enrollment[:email] : teacher[:email]
+
+  plp = DASHBOARD_DB[:professional_learning_partners].where(contact_id: workshop[:organizer_id]).first
+  # Have they ever taken any pd-workshop-survey?
+  is_first_survey = teacher_id.nil? || DB[:forms].where(kind: 'PdWorkshopSurveyCounselorAdmin', user_id: teacher_id).count == 0
+
+  # Have they taken this particular survey?
+  previous_response = DB[:forms].where(kind: 'PdWorkshopSurveyCounselorAdmin', source_id: enrollment[:id]).first
+
+  course = workshop[:course]
+  pass unless ['Counselor', 'Admin'].include? course
+
+  def local_view(name, options = {})
+    options[:form] = PdWorkshopSurveyCounselorAdmin
+    view "pd_workshop_survey/#{name.to_s}", options
+  end
+
+:javascript
+  window.pdWorkshopSurvey = {
+    enrollmentCode: "#{enrollment_code}",
+    course: "#{course}"
+  };
+
+%script{type: "text/javascript", src: "/js/sifter.min.js"}
+%script{type: "text/javascript", src: "/js/microplugin.min.js"}
+%script{type: "text/javascript", src: "/js/selectize.min.js"}
+%script{type: "text/javascript", src: "/js/selectize-fast-click.js"}
+%script{type: "text/javascript", src: "/js/pd-workshop-survey-counselor-admin.js"}
+%link{rel: "stylesheet", type: "text/css", href: "/css/selectize.bootstrap3.css"}/
+
+- if previous_response
+  %h1 Thank you
+  %h3 You have already completed this survey.
+
+- else
+
+  %form#pd-workshop-survey-form{role: 'form'}
+    %input{name: 'user_id_i', type: 'hidden', value: teacher_id}
+    %input{name: 'email_s', type: 'hidden', value: email}
+    %input{name: 'name_s', type: 'hidden', value: name}
+
+    %input{name: 'enrollment_id_i', type:'hidden', value: enrollment[:id]}
+    %input{name: 'workshop_id_i', type: 'hidden', value: workshop[:id]}
+    %input{name: 'plp_b', type: 'hidden', value: plp.nil? ? '0' : '1'}
+    %input{name: 'include_demographics_b', type: 'hidden', value: is_first_survey ? '1' : '0'}
```

Comment:
It doesn't look like `is_first_survey` and `include_demographics_b` are used here, and I don't see any demographic questions in the proposed list. This was used in the general survey to include demographic questions on the first survey response (i.e. first workshop) for a given teacher. See [haml](https://github.com/code-dot-org/code-dot-org/blob/staging/pegasus/sites.v3/code.org/public/pd-workshop-survey/splat.haml#L253-L293) and [form](https://github.com/code-dot-org/code-dot-org/blob/staging/pegasus/forms/pd_workshop_survey.rb#L66-L74).

I think these can be removed. 
If however we are to include the demographic questions we'll probably want to tweak the logic so that it's the first of either survey for any given teacher, rather than the first of the current survey  @type.